src/main/scala/circt/Convention.scala

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package circt
+
+import chisel3.experimental.{annotate, BaseModule, ChiselAnnotation}
+import firrtl.annotations.{ModuleTarget, SingleTargetAnnotation}
+import firrtl.FirrtlUserException
+
+/** Annotation to specify a module's port convention.
+  * The port convention defines how the ports of a module are transformed while
+  * lowering to verilog.
+  */
+case class ConventionAnnotation(target: ModuleTarget, convention: String) extends SingleTargetAnnotation[ModuleTarget] {
+  override def duplicate(target: ModuleTarget): ConventionAnnotation =
+    ConventionAnnotation(target, convention)
+}
```

Comment:
Two thoughts:

1. This can be made private. We only want people accessing it through the `convention` object.
2. This can actually be a `NoTargetAnnotation` instead of a `SingleTargetAnnotation[ModuleTarget]`. This avoids having to provide an implementation of the abstract `duplicate` method.

Reply:
Talked to @jackkoenig: there may be some issues with converting this to `NoTargetAnnotation` due to the injecting aspect path possibly relying on this. Let's keep this as `SingleTargetAnnotation[ModuleTarget]` for now.

Reply:
OK sounds good, leaving as a single target anno (this means I can't make `duplicate` private, because it's an overriden method :)) Thanks.